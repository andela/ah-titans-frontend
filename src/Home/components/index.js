import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import {
	Card, Col, Chip, Icon,
} from 'react-materialize';
import './index.scss';

const ArticlesForm = ({ article }) => (
	<Col s={12} m={6}>
		<Card
			key={article.slug}
			className="grey lighten-4 article"
			textClassName="black-text"
			title={article.title}
			actions={[
				<button className="btn waves-effect waves-light blue-grey">
					<i className="material-icons">share</i>
				</button>,
				<button className="btn waves-effect waves-light blue-grey">
					<i className="material-icons">thumb_up</i>
					<span>{article.likes_count}</span>
				</button>,
				<button className="btn waves-effect waves-light blue-grey">
					<i className="material-icons">thumb_down</i>
					<span>{article.dislikes_count}</span>
				</button>,
				<Link to={`/article/${article.slug}`}>
					<a className="right blue-grey-text" href="article link goes here">Read more...</a>
				</Link>,
			]}
		>
			{article.tagList && article.tagList.map(tag => <Chip>{tag}</Chip>)}

			<p className="ratings">
				<Icon blue>star</Icon>
				<i className="material-icons">star</i>
				<i className="material-icons icon-orange">star</i>
				<i className="material-icons icon-orange">star</i>
				<i className="material-icons icon-orange">star_half</i>
				<i style={{ alignContent: 'center', position: 'inline' }}>(4.6)</i>
				<i className="material-icons icon-blue right">
					{article.favorited ? 'favorite' : 'favorite_border'}
				</i>
			</p>

			<p />
			<Link to={`/article/${article.slug}`}>
				<p className="about">{article.description}</p>
				<br />
				<p style={{ color: 'black' }}>
					{article.body && article.body.substring(0, 150)}
					{article.body && article.body.length > 150 ? '...' : ''}
				</p>
			</Link>
			<hr className="light-grey" />
			<p className="writtenby">
          By:

								{' '}
				{article.author && article.author.username}
				<a href="url" className="follow">Follow</a>
				{moment(article.created_at).format('MMM Do YYYY, h:mm:ss a')}
			</p>
		</Card>
	</Col>
);

ArticlesForm.propTypes = {};

export default ArticlesForm;
