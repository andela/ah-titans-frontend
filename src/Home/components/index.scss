.icons {
  margin: 10px;
  border: 0px solid grey;
}
.writtenby {
  font-size: 12px;
}
.about {
  font-size: 16px;
  font-style: bold;
}
.home {
  margin-left: 50px;
}
.searchbar {
  margin-top: -20px;
  margin-left: -20px;
}
.pagination {
  text-align: center;
}
.follow {
  border: 0.5px solid white;
  border-radius: 5px;
  color: green;
  margin-left: 5px;
  margin-right: 175px;
  padding: 1px 5px 1px 5px;
}
.readmore {
  color: green;
}
.green {
  float: right;
  margin-right: 15px;
}
