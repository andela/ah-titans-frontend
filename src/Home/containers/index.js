import React, { Component } from 'react';
import SnackBar from 'react-material-snackbar';
import { connect } from 'react-redux';
import '../components/index.scss';
import ArticlesForm from '../components';
import getArticles from '../../actions/viewArticles';

class Home extends Component {
  componentDidMount() {
    this.props.getArticles();
  }
  render() {
    console.log(this.props.articles);
    const { new_user } = this.props.home;
    const { articles } = this.props.articles;
    return (
      <div className="app">
        {new_user ? (
          <SnackBar show timer={6000} className="home">
            You have successfully signed up to Authors Haven!! Please check your
            email to verify your account
          </SnackBar>
        ) : (
          ''
        )}

        <ArticlesForm articles={articles} />
      </div>
    );
  }
}

const mapStatetoProps = state => ({
  home: state.exampleReducer,
  articles: state.viewArticles,
});
export default connect(
  mapStatetoProps,
  { getArticles },
)(Home);
