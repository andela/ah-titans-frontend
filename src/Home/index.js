import React from 'react';
import Loader from '../views/Loader';

import './index.scss';

const Home = () => (
  <div className="app">
    <Loader />
    <h1>Welcome to Authors  Haven!</h1>
  </div>
);

export default Home;
