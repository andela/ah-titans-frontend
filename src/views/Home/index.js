import React from 'react';

const Home = () => (
  <div>
    <h1>Welcome to Authors  Haven!</h1>
  </div>
);

export default Home;
