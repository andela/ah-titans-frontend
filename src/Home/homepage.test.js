import React from 'react';
import { shallow, } from 'enzyme';
import Home from './components';

describe('Home', () => {
  it('renders without crashing', () => {
    shallow(<Home />);
  });
});
