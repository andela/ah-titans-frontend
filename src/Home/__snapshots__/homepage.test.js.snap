// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchComponent /> render() renders the component 1`] = `
<div
  className="row home"
>
  <Col
    m={10}
    s={12}
  >
    <div
      className="search-bar"
    >
      <Dropdown
        trigger={
          <Button
            node="button"
          >
            Filter By
            <i
              className="material-icons right"
            >
              filter_list
            </i>
          </Button>
        }
      >
        <NavItem>
          Tags
        </NavItem>
        <NavItem>
          Author
        </NavItem>
        <NavItem>
          Title
        </NavItem>
      </Dropdown>
      <div
        className="input-field"
      >
        <input
          id="search"
          required={true}
          type="search"
        />
        <label
          htmlFor="search"
        >
          <Icon>
            search
          </Icon>
        </label>
        <Icon>
          close
        </Icon>
      </div>
    </div>
  </Col>
</div>
`;
